### ISSUE TITLE
Excess type annotations are not errors/warnings

### ISSUE THREAD

**Issue by euclidianAce** [teal-language/tl#867]
This program unexpectedly compiles with no error or warning about the superfluous type annotation

```
local x: integer, string = 1
```

It's probably worth it to emit an error or warning when the number of types is greater than the number of variables declared.

The troublesome part of making this an error or warning could prevent some obscure corner case of using an annotation for bidirectional inference regarding generic function returns.

Even this contrived example is probably better off without the annotation but is _technically_ a use case:

```
global func: function<T, U>(U): T, U
local x: integer, string = func(1)
```

**Comment by hishamhm** [comment]
Oh, good call with the latter example there, though it might be simpler to treat the whole count mismatch as a bug and require that the user does `local x, _` there.

**Comment by hishamhm** [comment]
With the fix of #868 in 43def7f181c9ad297f8a92412c6654f4223f303d, this example does fail:

```
global func: function<T, U>(U): T, U
local x: integer, string = func(1)
```

but simply removing the second type does not fail:

```
global func: function<T, U>(U): T, U
local x: integer = func(1)
```

since nothing of type `U` is never used, it's not a problem that it is never resolved.

**Comment by hishamhm** [comment]
Fixed in https://github.com/teal-language/tl/commit/43def7f181c9ad297f8a92412c6654f4223f303d
